examples/async.rs

```diff
@@ -0,0 +1,293 @@
+    let token = std::env::var("GITHUB_TOKEN")?;
+    let crab = OctocrabBuilder::new().personal_token(token).build()?;
+    octocrab::initialise(crab);
+    Ok(())
+}
+
+#[derive(Debug, Default)]
+struct App {
+    should_quit: bool,
+    pulls: PullRequestsWidget,
+}
+
+impl App {
+    const FRAMES_PER_SECOND: f32 = 60.0;
+
+    pub async fn run(mut self, mut terminal: Terminal) -> Result<()> {
+        self.pulls.run();
+
+        let mut interval =
+            tokio::time::interval(Duration::from_secs_f32(1.0 / Self::FRAMES_PER_SECOND));
+        let mut events = EventStream::new();
+
+        while !self.should_quit {
+            tokio::select! {
+                _ = interval.tick() => self.draw(&mut terminal)?,
+                Some(Ok(event)) = events.next() =>  self.handle_event(&event),
+            }
+        }
+        Ok(())
+    }
+
+    fn draw(&self, terminal: &mut Terminal) -> Result<()> {
+        terminal.draw(|frame| {
+            let area = frame.size();
+            frame.render_widget(
+                Line::from("ratatui async example").centered().cyan().bold(),
+                area,
+            );
+            let area = area.offset(Offset { x: 0, y: 1 }).intersection(area);
+            frame.render_widget(&self.pulls, area);
+        })?;
+        Ok(())
+    }
+
+    fn handle_event(&mut self, event: &Event) {
+        if let Event::Key(event) = event {
+            match event.code {
+                KeyCode::Char('q') => self.should_quit = true,
+                KeyCode::Char('j') => self.pulls.scroll_down(),
+                KeyCode::Char('k') => self.pulls.scroll_up(),
+                _ => {}
+            }
+        }
+    }
+}
+
+/// A widget that displays a list of pull requests.
+///
+/// This is an async widget that fetches the list of pull requests from the GitHub API. It contains
+/// an inner `Arc<RwLock<PullRequests>>` that holds the state of the widget. Cloning the widget
+/// will clone the Arc, so you can pass it around to other threads, and this is used to spawn a
+/// background task to fetch the pull requests.
+#[derive(Debug, Clone, Default)]
+struct PullRequestsWidget {
+    inner: Arc<RwLock<PullRequests>>,
+    selected_index: usize, // no need to lock this since it's only accessed by the main thread
+}
+
+#[derive(Debug, Default)]
+struct PullRequests {
+    pulls: Vec<PullRequest>,
+    loading_state: LoadingState,
+}
+
+#[derive(Debug, Clone)]
+struct PullRequest {
+    id: String,
+    title: String,
+    url: String,
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+enum LoadingState {
+    #[default]
+    Idle,
+    Loading,
+    Loaded,
+    Error(String),
+}
+
+impl PullRequestsWidget {
+    /// Start fetching the pull requests in the background.
+    ///
+    /// This method spawns a background task that fetches the pull requests from the GitHub API.
+    /// The result of the fetch is then passed to the `on_load` or `on_err` methods.
+    fn run(&self) {
```

Comment:
Thoughts on renaming this to `spawn_fetches`? or something with `spawn` in the name?

Reply:
I'd prefer to keep this fairly standard / generic to indicate that this is something that would be found in most widgets / apps.